h/static/scripts/login-forms/components/test/SignupForm-test.js

```diff
@@ -0,0 +1,209 @@
+      email: 'Invalid email',
+      password: 'Invalid password',
+      privacy_accepted: 'You must accept the privacy policy',
+    };
+
+    const { elements } = createWrapper();
+    const { usernameField, emailField, passwordField, privacyCheckbox } =
+      elements;
+
+    assert.equal(
+      usernameField.prop('fieldError'),
+      fakeConfig.formErrors.username,
+    );
+    assert.equal(emailField.prop('fieldError'), fakeConfig.formErrors.email);
+    assert.equal(
+      passwordField.prop('fieldError'),
+      fakeConfig.formErrors.password,
+    );
+    assert.equal(
+      privacyCheckbox.prop('error'),
+      fakeConfig.formErrors.privacy_accepted,
+    );
+  });
+
+  it('pre-fills form fields from formData', () => {
+    fakeConfig.formData = {
+      username: 'testuser',
+      email: 'test@example.com',
+      password: 'testpassword',
+      privacy_accepted: true,
+      comms_opt_in: true,
+    };
+
+    const { elements } = createWrapper();
+    const {
+      usernameField,
+      emailField,
+      passwordField,
+      privacyCheckbox,
+      commsCheckbox,
+    } = elements;
+
+    assert.equal(usernameField.prop('value'), fakeConfig.formData.username);
+    assert.equal(emailField.prop('value'), fakeConfig.formData.email);
+    assert.equal(passwordField.prop('value'), fakeConfig.formData.password);
+    assert.equal(
+      privacyCheckbox.prop('checked'),
+      fakeConfig.formData.privacy_accepted,
+    );
+    assert.equal(
+      commsCheckbox.prop('checked'),
+      fakeConfig.formData.comms_opt_in,
+    );
+  });
+
+  it('updates username when input changes', () => {
+    const { wrapper, elements } = createWrapper();
+    const username = 'testuser';
+
+    act(() => {
+      elements.usernameField.prop('onChangeValue')(username);
+    });
+    wrapper.update();
+    const updatedElements = getElements(wrapper);
+
+    assert.equal(updatedElements.usernameField.prop('value'), username);
+  });
+
+  it('updates email when input changes', () => {
+    const { wrapper, elements } = createWrapper();
+    const email = 'test@example.com';
+
+    act(() => {
+      elements.emailField.prop('onChangeValue')(email);
+    });
+    wrapper.update();
+    const updatedElements = getElements(wrapper);
+
+    assert.equal(updatedElements.emailField.prop('value'), email);
+  });
+
+  it('updates password when input changes', () => {
+    const { wrapper, elements } = createWrapper();
+    const password = 'newpassword';
+
+    act(() => {
+      elements.passwordField.prop('onChangeValue')(password);
+    });
+    wrapper.update();
+    const updatedElements = getElements(wrapper);
+
+    assert.equal(updatedElements.passwordField.prop('value'), password);
+  });
+
+  it('updates privacy checkbox when clicked', () => {
+    const { wrapper, elements } = createWrapper();
+    const mockEvent = {
+      target: { checked: true },
+    };
+
+    act(() => {
+      elements.privacyCheckbox.prop('onChange')(mockEvent);
```

Comment:
I'll probably change this to set the real value of the checkbox and then dispatch the event.